xtext/editorserver/editor_tool.json

```diff
@@ -7,8 +7,8 @@
                 "id": "function-xtext-to-xmi",
                 "name": "xtext-to-xmi",
 
-                "parameters": [ {"name":"input", "type":"xtext-generated"}, 
-                                {"name":"language", "type":"text"}],
```

Comment:
Why is the `language` parameter not needed anymore? Doesn't that break the platform, which expects to set it by default?

Reply:
The language parameter is used by the platform for tool functions so that  a single function can handle multiple languages etl egl... etc. Conversion functions are specialised and perform only one function so they do not use the language parameter and just the parameter types are used for type conversion.

Reply:
So this means that `to-xmi` can now only be used as a conversion function? Does this break any of its current uses in activities?

Reply:
@szschaler  Yes that's right. None of the test activities I test with call xtext-to-xmi as an action function from a button. This highlights that our distinction between conversion functions and action functions is slightly unclear.  

To support this currently a separate tool function can be added in the tool config file. One entry for the conversion function without the language parameter and another for the action function with the language parameter. Different names need to be used an this creates some duplication in the config file.

If we want to make it possible to invoke a conversion function via a button, a button class e.g. ConversionButton should be added with a corresponding runConversion(...) function that actually invokes the remote tool service. Similar to runAction() but without the type conversion.